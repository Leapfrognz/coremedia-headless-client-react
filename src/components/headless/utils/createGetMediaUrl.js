// @flow
const createGetMediaUrl = (baseURL: string, prefix: string) => (
  link: string,
  ratio?: string,
  minWidth?: number
): string => {
  if (link && link.startsWith(prefix)) {
    const parts = link.substring(prefix.length).split('/');
    if (parts.length === 3) {
      const [type, id, property] = parts;
      switch (type) {
        case 'image': {
          const mediaUrl = `${baseURL}/media/${id}/${property}`;
          let queryParams = '';
          if (ratio) {
            queryParams = `?ratio=${ratio}`;
          }
          if (minWidth) {
            queryParams = `${!queryParams ? '?' : `${queryParams}&`}minWidth=${minWidth}`;
          }
          return mediaUrl + queryParams;
        }
        case 'media': {
          return baseURL + `/media/${id}/${property}`;
        }
        default:
          return '';
      }
    }
  }
  return '';
};

export default createGetMediaUrl;
