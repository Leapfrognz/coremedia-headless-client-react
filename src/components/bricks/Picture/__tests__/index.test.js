// @flow
import React from 'react';

import createGetMediaUrl from '../../../headless/utils/createGetMediaUrl';

describe('Picture Component', () => {
  const context = {
    default: {
      getMediaUrl: link => link,
    },
    headless: {
      getMediaUrl: createGetMediaUrl(
        'http://127.0.0.1:8080/caas/v1/coremedia/sites/caassiopeia-en-DE',
        'coremedia:///'
      ),
    },
  };

  const getPictureWithContext = (variant = 'default') => {
    jest.doMock('../../../headless/Context/MediaContext', () => {
      return {
        Consumer: props => props.children(context[variant]),
      };
    });

    return require('..').default;
  };

  describe('default context', () => {
    it('should render correctly', () => {
      const Picture = getPictureWithContext();
      const wrapper = shallow(
        <Picture
          color="red"
          link="http://127.0.0.1:8080/media/2662/data"
          ratio="portrait_ratio1x1"
          title="Scrum"
          theme={{
            breakpoints: {
              small: 480,
              tablet: 768,
              desktop: 992,
              large: 1280,
            },
          }}
        />
      );
      expect(
        wrapper
          .shallow()
          .find('Consumer')
          .shallow()
      ).toMatchSnapshot();
    });
  });
  describe('headless context', () => {
    it('should render correctly', () => {
      const Picture = getPictureWithContext('headless');
      const wrapper = shallow(
        <Picture
          color="red"
          link="coremedia:///image/2662/data"
          ratio="portrait_ratio1x1"
          title="Scrum"
          theme={{
            breakpoints: {
              small: 480,
              tablet: 768,
              desktop: 992,
              large: 1280,
            },
          }}
        />
      );
      expect(
        wrapper
          .shallow()
          .find('Consumer')
          .shallow()
      ).toMatchSnapshot();
    });
    it('should render correctly without <sources> element', () => {
      const Picture = getPictureWithContext('headless');
      const wrapper = shallow(
        <Picture
          color="red"
          link="coremedia:///image/2662/data"
          ratio="portrait_ratio1x1"
          title="Scrum"
          theme={{}}
        />
      );
      expect(
        wrapper
          .shallow()
          .find('Consumer')
          .shallow()
      ).toMatchSnapshot();
    });
    it('should render correctly with ratio thumbnail', () => {
      const Picture = getPictureWithContext('headless');
      const wrapper = shallow(
        <Picture
          color="red"
          link="coremedia:///image/2662/data"
          ratio="thumbnail"
          title="Scrum"
          theme={{
            breakpoints: {
              small: 480,
              tablet: 768,
              desktop: 992,
              large: 1280,
            },
          }}
        />
      );
      expect(
        wrapper
          .shallow()
          .find('Consumer')
          .shallow()
      ).toMatchSnapshot();
    });
    it('should render correctly and skip breakpoint large', () => {
      const Picture = getPictureWithContext('headless');
      const wrapper = shallow(
        <Picture
          color="red"
          link="coremedia:///image/2662/data"
          ratio="portrait_ratio1x1"
          title="Scrum"
          theme={{
            breakpoints: {
              small: 480,
              tablet: 768,
              desktop: 992,
              large: '1280',
            },
          }}
        />
      );
      expect(
        wrapper
          .shallow()
          .find('Consumer')
          .shallow()
      ).toMatchSnapshot();
    });
  });
});
