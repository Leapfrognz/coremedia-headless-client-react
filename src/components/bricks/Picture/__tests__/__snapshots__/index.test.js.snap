// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture Component default context should render correctly 1`] = `
<picture>
  <source
    key="0"
    media="(min-width: 1280px)"
    srcSet="http://127.0.0.1:8080/media/2662/data"
  />
  <source
    key="1"
    media="(min-width: 992px)"
    srcSet="http://127.0.0.1:8080/media/2662/data"
  />
  <source
    key="2"
    media="(min-width: 768px)"
    srcSet="http://127.0.0.1:8080/media/2662/data"
  />
  <source
    key="3"
    media="(min-width: 480px)"
    srcSet="http://127.0.0.1:8080/media/2662/data"
  />
  <BasicImg
    src="http://127.0.0.1:8080/media/2662/data"
    title="Scrum"
  />
</picture>
`;

exports[`Picture Component headless context should render correctly 1`] = `
<picture>
  <source
    key="0"
    media="(min-width: 1280px)"
    srcSet="coremedia:///image/2662/data"
  />
  <source
    key="1"
    media="(min-width: 992px)"
    srcSet="coremedia:///image/2662/data"
  />
  <source
    key="2"
    media="(min-width: 768px)"
    srcSet="coremedia:///image/2662/data"
  />
  <source
    key="3"
    media="(min-width: 480px)"
    srcSet="coremedia:///image/2662/data"
  />
  <BasicImg
    src="coremedia:///image/2662/data"
    title="Scrum"
  />
</picture>
`;

exports[`Picture Component headless context should render correctly and skip breakpoint large 1`] = `
<picture>
  <source
    key="0"
    media="(min-width: 992px)"
    srcSet="coremedia:///image/2662/data"
  />
  <source
    key="1"
    media="(min-width: 768px)"
    srcSet="coremedia:///image/2662/data"
  />
  <source
    key="2"
    media="(min-width: 480px)"
    srcSet="coremedia:///image/2662/data"
  />
  <BasicImg
    src="coremedia:///image/2662/data"
    title="Scrum"
  />
</picture>
`;

exports[`Picture Component headless context should render correctly with ratio thumbnail 1`] = `
<BasicImg
  src="coremedia:///image/2662/data"
  title="Scrum"
/>
`;

exports[`Picture Component headless context should render correctly without <sources> element 1`] = `
<picture>
  <BasicImg
    src="coremedia:///image/2662/data"
    title="Scrum"
  />
</picture>
`;
